pkg/services/object/get/util.go

```diff
@@ -222,72 +219,50 @@ func (c *clientWrapper) get(exec *execCtx, key *ecdsa.PrivateKey) (*object.Objec
 
 	hdr, rdr, err := c.client.ObjectGetInit(exec.context(), addr.Container(), id, user.NewAutoIDSigner(*key), opts)
 	if err != nil {
-		return nil, fmt.Errorf("init object reading:: %w", err)
+		return nil, nil, fmt.Errorf("init object reading:: %w", err)
 	}
-
-	buf := make([]byte, hdr.PayloadSize())
-
-	_, err = rdr.Read(buf)
-	if err != nil && !errors.Is(err, io.EOF) {
-		return nil, fmt.Errorf("read payload: %w", err)
-	}
-
-	hdr.SetPayload(buf)
-
-	return &hdr, nil
+	return &hdr, rdr, nil
 }
 
-func (e *storageEngineWrapper) get(exec *execCtx) (*object.Object, error) {
+func (e *storageEngineWrapper) get(exec *execCtx) (*object.Object, io.ReadCloser, error) {
 	if exec.headOnly() {
 		r, err := e.engine.Head(exec.address(), exec.isRaw())
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
-		return r, nil
+		return r, nil, nil
 	}
 
 	if rng := exec.ctxRange(); rng != nil {
 		r, err := e.engine.GetRange(exec.address(), rng.GetOffset(), rng.GetLength())
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		o := object.New()
 		o.SetPayload(r)
 
-		return o, nil
+		return o, nil, nil
```

Comment:
we return stream only in remote case. Suggest to do the same here. Although `engine.GetRange` is not streamed yet, we can prepare for this in advance